pkg/pillar/README.md

```diff
@@ -42,3 +43,24 @@ Make targets of note:
 - `make builder-image`: build the builder image for your user
 - `make shell`: launch a shell inside the builder image, whence you can run all commands. It also sets `BUILD=local`, so that you can run just `make build` inside.
 
+## States and their transitions
+
+Object inside agents (more precisely inside [pubsub](./pubsub/doc.go)) may have two descriptions:
+
+1. ObjectConfig - description of object comes directly from config or generated during parsing of config
+1. ObjectStatus - description of object generated by agent as a current state of processing of ObjectConfig
+
+Those structures are described in [types](./types). Agents have publishers and subscribers via pubsub to types from
+other agents. ObjectStatus may have several states inside, generally,
+they have `SwState`, but may have additional booleans or `SubStates` described current state of object.
+Agents obtain ObjectConfig and do their work by creating ObjectStatus in Initial state and moving it to another values.
+ObjectStatus is sending through pubsub to another agents, which have subscribes, and actions are fired in them depending
+on the State. State transitions of the particular ObjectStatus should be written in one place (in one function,
+you can find several with names starts with doUpdate).
+
+Actions may be sync (they must be executable in limited time and not depends on another, for example,
+check file or directory existence, touch file, make directory and so on) and async. The second variant is done by
+[worker](./worker/doc.go) package. Worker instance has limited job queue and parallel workers in pool.
```

Comment:
The carridge returns in these paragraphs causes line breaks in the display of the file.
Better to have each paragraph be a single line to make it render uniformly

Reply:
I reduce line count. I think, too long lines is also not good idea (I cannot understand the problem in markdown file with carriage return), so few lines per paragraph.